SherpaBackEnd/Controllers/SurveyController.cs

```diff
@@ -1,6 +1,64 @@
+using Microsoft.AspNetCore.Mvc;
+using SherpaBackEnd.Dtos;
+using SherpaBackEnd.Exceptions;
+using SherpaBackEnd.Model.Survey;
+using SherpaBackEnd.Services;
+
 namespace SherpaBackEnd.Controllers;
 
+[ApiController]
+[Route("[controller]")]
 public class SurveyController
 {
-    
+    private readonly ISurveyService _surveyService;
+    private readonly ILogger _logger;
+
+    public SurveyController(ISurveyService surveyService, ILogger<SurveyController> logger)
+    {
+        _surveyService = surveyService;
+        _logger = logger;
+    }
+
+    [HttpPost]
+    public async Task<ActionResult> CreateSurvey(CreateSurveyDto createSurveyDto)
+    {
+        try
+        {
+            await _surveyService.CreateSurvey(createSurveyDto);
+
+            return new CreatedResult("", null);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(default, e, e.Message);
+
+            return e switch
+            {
+                NotFoundException => new ObjectResult(e)
```

Comment:
NotFoundException? I believe this is a 404 aren't we looking for a BadRequest here? Line 58 of this file looks for a a NotFoundException => new ObjectResult(e)
                    { StatusCode = StatusCodes.Status404NotFound, Value = e.Message },

Reply:
Yep, this is bc if there is a not found exception is because the team or the template that the user want to associate to that survey doesn't exist, so their request is incorrect, that's why we put a 400